samcli/commands/_utils/click_mutex.py

```diff
@@ -0,0 +1,62 @@
+"""
+Module to check mutually exclusive cli parameters
+"""
+from typing import Any, List, Dict, Tuple
+
+import click
+
+
+class Mutex(click.Option):
+    """
+    Preprocessing checks for mutually exclusive or required parameters as supported by click api.
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.required_param_lists: List = kwargs.pop("required_param_lists", None)
```

Comment:
For a list type, should default value be `None` or an empty list `[]`? We don't need to check if it is `None` when we are looping.

Reply:
Updated to "" and []. However we still need to check `if self.required_param_lists:` to skip over the `for` loop `else` clause.